base/loading.jl

```diff
@@ -2819,11 +2819,20 @@ global mkpidlock_hook
 global trymkpidlock_hook
 global parse_pidfile_hook
 
-# allows processes to wait if another process is precompiling a given source already
-function maybe_cachefile_lock(f, pkg::PkgId, srcpath::String)
+# The preferences hash is only known after precompilation so just assume no preferences
+# meaning that if all other conditions are equal, the same package cannot be precompiled
+# with different preferences at the same time.
+compilecache_pidfile_path(pkg::PkgId) = compilecache_path(pkg, UInt64(0)) * ".pidfile"
```

Comment:
This also uses the path to `Base.active_project`, which is also undesirable for the same reason as preferences. Can you make that an argument there, and pass `""` here to ignore it for this lock?

Reply:
Changed in https://github.com/JuliaLang/julia/pull/50254